pkg/pillar/diskmetrics/usage.go

```diff
@@ -18,6 +19,17 @@ import (
 	"github.com/shirou/gopsutil/disk"
 )
 
+// StatAllocatedBytes returns the allocated size of a file in bytes
+// This value will be less than fileInfo.Size() for thinly allocated files
+func StatAllocatedBytes(path string) (uint64, error) {
+	var stat syscall.Stat_t
+	err := syscall.Stat(path, &stat)
+	if err != nil {
+		return uint64(0), err
+	}
+	return uint64(stat.Blocks * int64(stat.Blksize)), nil
```

Comment:
Will this do the right thing on ZFS as well? (I never looked at what blocksize it returns.)

Reply:
ZFS path instead goes through a the zvol volume handler and calls libzfs module for values.